library/src/main/java/com/getstream/sdk/chat/adapter/viewholder/message/configurators/MessageTextConfigurator.kt

```diff
@@ -160,10 +161,13 @@ internal class MessageTextConfigurator(
         message: Message
     ) {
         binding.tvText.setOnClickListener {
-            if (message.isFailed() && !ChatClient.instance().isSocketConnected()) {
-                return@setOnClickListener
+            if (message.isFailed()) {
+                if (ChatClient.instance().isSocketConnected()) {
+                    messageRetryListener.onRetryMessage(message)
+                }
+            } else {
+                messageClickListener.onMessageClick(message)
```

Comment:
The connection logic should be out of the view
```suggestion
            if (message.isFailed()) {
                    messageRetryListener.onRetryMessage(message)
            } else {
                messageClickListener.onMessageClick(message)
```

On the other hand, why don't we attach the listener only on the "Message wasn't sent, try again" text?

Reply:
I removed the network check. Tested it, and the UX is pretty okay even if we don't have a connection when we perform the retry.

Also improved the condition, it now matches the condition for displaying the tap-to-retry text exactly.